src/record_syscall.cc

```diff
@@ -2556,6 +2577,20 @@ static Switchable prepare_ptrace(RecordTask* t,
       }
       break;
     }
+    case PTRACE_PEEKTEXT:
+    case PTRACE_PEEKDATA:
+    case PTRACE_POKETEXT:
+    case PTRACE_POKEDATA:
+    case PTRACE_PEEKUSER:
+    case PTRACE_POKEUSER:
+    case PTRACE_GETREGS:
+    case PTRACE_GETFPREGS:
+    case PTRACE_GETFPXREGS:
+    case PTRACE_SETREGS:
+    case PTRACE_SETFPREGS:
+    case PTRACE_SETFPXREGS:
+      prepare_ptrace_legacy<Arch>(t, syscall_state);
```

Comment:
I didn't check very hard that you copy/pasted the code here correctly :)

Reply:
Well, I didn't touch it, but hopefully the tests would catch it if I had ;). I did catch #2559 while doing this copy pasting so hopefully it's actually better than before :).